app/controllers/util/CustomMetadataUtils.scala

```diff
@@ -0,0 +1,112 @@
+package controllers.util
+
+import graphql.codegen.GetCustomMetadata.customMetadata.CustomMetadata
+import graphql.codegen.types.DataType
+import graphql.codegen.types.DataType._
+import graphql.codegen.types.PropertyType.{Defined, Supplied}
+
+import scala.collection.immutable.ListSet
+
+class CustomMetadataUtils(allCustomMetadataProperties: List[CustomMetadata]) {
+  private val allCustomMetadataPropertiesByName: Map[String, List[CustomMetadata]] = allCustomMetadataProperties.groupBy(_.name)
+
+  def getCustomMetadataProperties(propertiesToGet: Set[String]): Set[CustomMetadata] =
+    propertiesToGet.flatMap(property => allCustomMetadataPropertiesByName(property))
+
+  def getValuesOfProperties(namesOfPropertiesToGetValuesFrom: Set[String]): Map[String, List[CustomMetadata.Values]] = {
+    val propertiesToGetValuesFrom: Set[CustomMetadata] = getCustomMetadataProperties(namesOfPropertiesToGetValuesFrom)
+    propertiesToGetValuesFrom.map(property => property.name -> property.values).toMap
+  }
+
+  def convertPropertiesToFields(dependencyProperties: Set[CustomMetadata]): Set[(FieldValues, String)] =
+    dependencyProperties.map {
+      dependencyProperty => {
+        val (options, selectedOption) = generateFieldOptions(dependencyProperty, dependencyProperty.dataType)
+        FieldValues(
+          dependencyProperty.name,
+          fieldOptions=options,
+          selectedFieldOption=selectedOption,
+          multiValueSelect=if(dependencyProperty.multiValue) true else false,
+          fieldLabel=convertDbNameToFieldLabel(dependencyProperty.fullName.get),
```

Comment:
Ah, I see. Cool, I'll make it return an empty string with the assumption that if was supposed to have a full name, someone will add it because the `name` is a camelCased version of the full name, which looks weird and if someone sees an empty value, it might prompt them to add a full name instead of leaving it the way it is.